configure.ac

```diff
@@ -507,6 +507,37 @@ AC_MSG_CHECKING([sould NCP support UART])
 AC_MSG_RESULT([${OPENTHREAD_ENABLE_UART}])
 
 #
+# Single Instance Optimization
+#
+
+AC_ARG_ENABLE(single-instance-opt,
```

Comment:
We already have `OPENTHREAD_MULTIPLE_INSTANCE` as an option. It's mainly used for Windows platform and we check it with `#ifdef` instead of `#if` (would be good to change this to follow `#if` model later and also add it as a `configure.ac` option).

Here is how I see this:
 - `OPENTHREAD_MULTIPLE_INSTANCE` (if defined) determines if we have multiple ot instance vs a single instance (default case we don't have this defined, so we are single-instance). 
- `OPENTHREAD_ENABLE_SINGLE_INSTANCE_OPTIMIZATION` is used to enable the optimization feature.

With this model we can have single-instance case with or without the optimization enabled. 

Thoughts?